byway/src/main/webapp/recommendations.js

```diff
@@ -164,14 +165,32 @@ function findPlacesWithTextSearch(request, statuses) {
  * @param {Set} statuses String elements with status codes from placesService
  */
 function alertUser(statuses) {
-  document.getElementById("message-container").style.visibility = 'visible';
-  document.getElementById("general-message").style.visibility = 'visible';
-  let statusesContainer = document.getElementById("statuses");
-  statusesContainer.style.visibility = 'visible';
-  for(let status of statuses) {
-    let statusElement = document.createElement('ul');
-    statusElement.innerText = status;
-    statusesContainer.appendChild(statusElement);
+  let msgContainer = document.getElementById("message-container");
+  msgContainer.style.visibility = 'visible';
+  const allStatuses = formatStatusMessages(statuses);
+  msgContainer.innerText = "Showing limited results due to: " + allStatuses;
+}
+
+/**
+ * Go through status messages and add punctuation to make a sentence.
+ * @param {Set} statuses contains elements with status codes from placesService
+ * @return String with punctuation as it lists all status codes.
+ */
+function formatStatusMessages(statuses) {
```

Comment:
Trying to add proper punctation to the individual status messages is not necessary and adds clutter to your code. You could do something much simpler like delineate them with new lines. e.g.

```javascript
errorsStr = Array.from(statuses).join("\n")
msgContainer.innerText = "Showing limited results due to: \n" + errorsStr;
```

Reply:
Thanks for the suggestion, fixed